assets/app/views/images.html

```diff
@@ -1,17 +1,45 @@
 <div ng-controller="ProjectController" class="content">
   <project-page>
     <div ng-controller="ImagesController">
-      <h1>Images</h1>
+      <h1>Image Streams</h1>
       <alerts alerts="alerts"></alerts>      
-      <div ng-if="(images | hashSize) == 0">
+      <div ng-if="(imageStreams | hashSize) == 0">
         <div>
           <em>{{emptyMessage}}</em>
         </div>
       </div>      
-      <div style="margin-bottom: 10px;" ng-repeat="image in images">
-        <h3>{{image.dockerImageReference | imageName}} <span class="small">({{image.metadata.name}})</span></h3>
-        <div>Created: <relative-timestamp timestamp="image.metadata.creationTimestamp"></relative-timestamp></div>
-        <div ng-if="build = (image | buildForImage : builds)">Created from: Build {{build.metadata.labels.buildconfig}} (<a href="project/{{projectName}}/browse/builds" class="small" title="{{build.metadata.name}}">{{build.metadata.name.substr(0, 10)}}</a>)
+      <div ng-repeat="imageStream in imageStreams">
+        <div class="tile" ng-repeat="tag in imageStream.status.tags">
+          <div>
+            <span>{{imageStream.metadata.name}}:{{tag.tag}} &mdash; </span>
+            <span ng-if="!tag.items.length"><em>No images for this tag</em></span>
+            <span ng-if="tag.items.length">
+              <short-id ng-if="tag.items[0].image" id="{{tag.items[0].image}}"></short-id>
+              <span ng-if="!tag.items[0].image">{{tag.items[0].dockerImageReference}}</span>
```

Comment:
> if you change an imageStream's spec.tags[tag] to modify it's dockerImageReference, what happens to the dockerImageReference field in existing status.tags[tag].items?

The entry in status.tags[tag].items[0] will be updated with the new dockerImageReference; it will not add a new entry

> if you change an imageStream's spec.tags[tag] to change from an image to a dockerImageReference (or vice-versa), what happens to the dockerImageReference and image fields in existing status.tags[tag].items?

Here are the rules that govern status.tags:

1. when you use dockerImageReference in spec.tags, we never have an image for status.tags
1. when you use from to point at an image stream tag or image, we may or may not have an image (it depends on the target)
1. if both status.tags[tag].items[0].dockerImageReference and status.tags[tag].items[0].image change (i.e. new pull-by-id reference), we prepend a new entry to items
1. if only 1 of dockerImageReference or image changes, we modify the existing item

> Also, does changing from image to dockerImageReference (or modifying the dockerImageReference) in the spec trigger builds and deployment configs to pick up the new image?

Builds and deployments only trigger off of dockerImageReference. If it changes, they'll trigger.